### PR TITLE
chapter 3: static page

### PR DESCRIPTION
**Pull request by ngyihkai**

### CODE REVIEW

**Comment by ngyihkai** [comment]
Ready

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (huongnt-2545) [inline app/helpers/application_helper.rb]: apply I18n to translate this text
ex:
```suggestion
    base_title = t "rails_app_title"
```

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+  <header class="navbar navbar-fixed-top navbar-inverse">
+    <div class="container">
+      <%= link_to "sample app", "#", id: "logo" %>
```
Comment (huongnt-2545) [inline app/views/layouts/_header.html.erb]: use I18n for all text in html file
ex:
```suggestion
      <%= link_to t("sample_app"), "#", id: "logo" %>
```

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -11,7 +10,11 @@
   </head>
 
   <body>
+    <%= render "layouts/header" %>
+  <div class="container">
```
Comment (huongnt-2545) [inline app/views/layouts/application.html.erb]: indent line 14-17
```suggestion
    <div class="container"><%= yield %></div>
  </body>
```

File: app/views/static_pages/help.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<% provide :title, "Help" %>
```
Comment (huongnt-2545) [inline app/views/static_pages/help.html.erb]: I18n for all text in this file
ex:
```suggestion
<% provide :title, t("help") %>
```

File: app/views/static_pages/home.html.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<% provide :title, "Home" %>
```
Comment (huongnt-2545) [inline app/views/static_pages/home.html.erb]: I18n same

File: test/fixtures/users.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huongnt-2545) [inline test/fixtures/users.yml]: Don't do all test practice in the tutorial, it is no need
you can delete test file

**Comment by ngyihkai** [comment]
Ready

File: config/locales/en.yml [hunk 7]
```diff
@@ -1,34 +1,2 @@
...
-# en:
-#   'true': 'foo'
-#
-# To learn more, please read the Rails Internationalization guide
-# available at https://guides.rubyonrails.org/i18n.html.
-
-# config/locales/en.yml
 en:
-  titile: "Sample App"
+  title: "Sample App"
```
Comment (huongnt-2545) [inline config/locales/en.yml:2]: you need define all text at html, helper, controller, ... files apply I18n here
I don't see it

File: app/views/static_pages/home.html.erb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<% provide :title, t("Home")  %>
+
+<h1>Sample App</h1>
```
Comment (huongnt-2545) [inline app/views/static_pages/home.html.erb]: apply I18n for all text in this file, and define it on `locales/en.yml` file

File: app/views/static_pages/help.html.erb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<% provide :title, t("help")  %>
+
+<h1>Help</h1>
```
Comment (huongnt-2545) [inline app/views/static_pages/help.html.erb]: apply I18n for all text in this file

**Comment by ngyihkai** [comment]
Ready

File: test/models/user_test.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (ngocvt-0484) [inline test/models/user_test.rb]: you don't need do test
please delete this file

File: app/views/static_pages/help.html.erb [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<% provide :title, "Help" %>
+
+<h1><%= t :help %></h1>
+<p>
+  <%= t :help_1 %>
+  <a href="http://www.railstutorial.org/help"><%= t :help_2 %></a>
+  <%= t :help_3 %>
+  <a href="http://www.railstutorial.org/book"><%= t :help_4 %></a>.
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: please use link_to

File: app/views/static_pages/help.html.erb [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+</p>
+
+
+
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: delete empty line

File: app/views/layouts/application.html.erb [hunk 13]
```diff
@@ -11,7 +10,8 @@
   </head>
 
   <body>
-    <%= yield %>
-  </body>
+    <%= render "layouts/header" %>
+  <div class="container"><%= yield %>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb]:
```suggestion
  <%= render "layouts/header" %>
  <div class="container"><%= yield %></div>
```

Comment (ngocvt-0484) [reply app/views/layouts/application.html.erb]: it should align

**Comment by ngyihkai** [comment]
Ready

File: app/views/static_pages/help.html.erb (hunk 4 above)
Comment (ghost) [reply app/views/static_pages/help.html.erb]: fix this comment please

File: app/views/static_pages/home.html.erb (hunk 5 above)
Comment (ghost) [reply app/views/static_pages/home.html.erb]: fix this comment please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ngyihkai** [comment]
Ready

File: app/views/static_pages/help.html.erb [hunk 14]
```diff
@@ -0,0 +1,9 @@
+<% provide :title, t :Help %>
+
+<h1><%= t :Help %></h1>
+<p>
+  <%= link_to t(:help_1) %>
+  <a href="http://www.railstutorial.org/help"><%= link_to t(:help_2) %></a>
+  <%= link_to t(:help_3) %>
+  <a href="http://www.railstutorial.org/book"><%= link_to t(:help_4) %></a>.
```
Comment (ghost) [inline app/views/static_pages/help.html.erb]: please read https://apidock.com/rails/ActionView/Helpers/UrlHelper/link_to
and study to use link_to

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ngyihkai** [comment]
Ready

File: app/views/static_pages/help.html.erb [hunk 15]
```diff
@@ -0,0 +1,9 @@
+<% provide :title,  :Help %>
+
+<h1><%= t :Help %></h1>
+<p>
+  <%= link_to t(:help_1) %>
```
Comment (ghost) [inline app/views/static_pages/help.html.erb]: fix link_to in this file

File: app/views/static_pages/home.html.erb [hunk 16]
```diff
@@ -0,0 +1,10 @@
+<% provide :title,  :Home %>
+
+<h1><%= t :Sample_App %></h1>
+<p>
+  <%= t :p_1 %>
+  <a href="http://www.railstutorial.org/">
```
Comment (ghost) [inline app/views/static_pages/home.html.erb]: use link_to please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ngyihkai** [comment]
Ready

File: app/views/static_pages/help.html.erb [hunk 17]
```diff
@@ -0,0 +1,9 @@
+<% provide :title,  :Help %>
```
Comment (ghost) [inline app/views/static_pages/help.html.erb]:
```suggestion
<% provide :title, :Help %>
```

File: app/assets/stylesheets/custom.scss [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+  letter-spacing: -1px;
+  padding-top: 9px;
+  font-weight: bold;
+}
+
+#logo:hover {
+  color: #fff;
+  text-decoration: none;
+}
+
```
Comment (ghost) [inline app/assets/stylesheets/custom.scss]: remove this blank line

File: app/views/static_pages/home.html.erb [hunk 19]
```diff
@@ -0,0 +1,10 @@
+<% provide :title, :Home %>
+
+<h1><%= t :Sample_App %></h1>
+<p>
+  <%= t :p_1 %>
+  <%= link_to t(:p_2), "http://www.railstutorial.org/" %>
+  <%= t :p_2 %>
+  <%= t :p_3 %>
+</p>
+
```
Comment (ghost) [inline app/views/static_pages/home.html.erb]: remove this blank line

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ngyihkai** [comment]
Ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**